pkg/3scale/amp/component/system.go

```diff
@@ -241,6 +249,24 @@ func (system *System) buildSystemBaseEnv() []v1.EnvVar {
 	return result
 }
 
+func (system *System) buildAppMasterContainerEnv() []v1.EnvVar {
+	result := system.buildSystemBaseEnv()
+	result = append(result, helper.EnvVarFromValue(SystemAppPrometheusExporterPortEnvVarName, strconv.Itoa(SystemAppMasterContainerPrometheusPort)))
```

Comment:
I'm not sure whether the definition of this environment variable should be conditional to the appMetrics boolean. We could do it but because there are several metrics ports in the same pod between containers leaving the default might cause port collision between containers in the templates approach if we don't include the environment variables?

On the other hand, kind of weird not including envvars conditionally but doing it for the containerports.

What do you think?

Reply:
We need to write an ugrade procedure as they are not reconciled automatically